### PR TITLE
extensions: fix `enchant` extension for PHP < 8

### LINKED ISSUE #247: `enchant` extension is broken
This is broken:

- `nix build .#php56.extensions.enchant`
- `nix build .#php70.extensions.enchant`
- `nix build .#php71.extensions.enchant`
- `nix build .#php72.extensions.enchant`
- `nix build .#php73.extensions.enchant`
- `nix build .#php74.extensions.enchant`

This has been found while working on https://github.com/loophp/nix-shell/pull/157 and https://github.com/fossar/nix-phps/pull/246

### PR DESCRIPTION
**Pull request by drupol**
This PR

- Should fix #247

Source of `enchant1` derivation: https://github.com/matthewbauer/nixpkgs/blob/ba76fa91c762160e0d75a4cba74b44a0c3f911e7/pkgs/development/libraries/enchant/1.x.nix

### CODE REVIEW

File: pkgs/package-overrides.nix [hunk 1]
```diff
@@ -149,6 +149,21 @@ in
         attrs.preConfigure or "" + linkInternalDeps deps;
     });
 
+    enchant = let
+      enchant1 = pkgs.callPackage ./enchant/1.x.nix { };
+    in
+    prev.extensions.enchant.overrideAttrs (attrs: {
+      nativeBuildInputs = attrs.nativeBuildInputs or [] ++
+        lib.optionals (lib.versionOlder prev.php.version "8.0") [
+          enchant1 pkgs.pkg-config
```
Comment (jtojnar) [inline pkgs/package-overrides.nix:158]: enchant would have to go to `buildInputs`.

File: pkgs/enchant/1.x.nix [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+stdenv.mkDerivation (finalAttrs: {
+  version = "1.6.1";
+  pname = "enchant";
+
+  src = fetchurl {
+    url = "https://github.com/AbiWord/enchant/releases/download/enchant-${finalAttrs.version}/enchant.tar.gz";
+    sha256 = "1xg3m7mniyqyff8qv46jbfwgchb6di6qxdjnd5sfir7jzv0dkw5y";
+  };
+
+  nativeBuildInputs = [ pkgconfig ];
```
Comment (jtojnar) [inline pkgs/enchant/1.x.nix:12]: `pkgconfig` is an alias.

File: pkgs/enchant/1.x.nix [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  buildInputs = [ aspell glib hunspell hspell ];
+
+  meta = {
+    description = "Generic spell checking library";
+    homepage = "https://abiword.github.io/enchant";
+    platforms = lib.platforms.unix;
+    license = lib.licenses.lgpl21;
```
Comment (jtojnar) [inline pkgs/enchant/1.x.nix:19]: `lgpl21` is unclear, use `lgpl21Plus` or `lgpl21Only`. See https://discourse.nixos.org/t/lib-licenses-gpl3-co-are-now-deprecated/8206 for more details.

File: pkgs/enchant/1.x.nix (hunk 2 above)
Comment (jtojnar) [inline pkgs/enchant/1.x.nix:12]: We might as well do one dependency per line.

File: pkgs/enchant/1.x.nix [hunk 4]
```diff
@@ -0,0 +1,21 @@
+{ stdenv, lib, fetchurl, aspell, pkgconfig, glib, hunspell, hspell }:
```
Comment (jtojnar) [inline pkgs/enchant/1.x.nix:1]: I do not really like keeping five year old packages. Ideally we would extract patches from the history https://github.com/php/php-src/commits/master/ext/enchant and bump our extension to enchant 2 but not sure it is worth the effort.

Comment (drupol) [reply pkgs/enchant/1.x.nix]: I understand. In this case, `enchant2` is only [supported from PHP 8](https://www.php.net/manual/en/enchant.requirements.php), so I guess this PR is maybe the best option. No?

File: pkgs/enchant/1.x.nix (hunk 2 above)
Comment (drupol) [reply pkgs/enchant/1.x.nix]: Done. Replaced with `pkg-config`.

File: pkgs/enchant/1.x.nix (hunk 3 above)
Comment (drupol) [reply pkgs/enchant/1.x.nix]: Done, replaced with `lgpl21Only`.

File: pkgs/enchant/1.x.nix (hunk 4 above)
Comment (jtojnar) [reply pkgs/enchant/1.x.nix]: Well, we could make it supported in older versions, like we are doing with OpenSSL 1.1. But I am not sure it is worth it. I doubt anyone even uses enchant from PHP.

**Review by jtojnar — APPROVED** [review]

**Pull request merged**